cpp/vm/evmzero/interpreter_test.cc

```diff
@@ -5198,6 +5198,124 @@ TEST(InterpreterTest, Step_Single_Memory) {
       1);
 }
 
+///////////////////////////////////////////////////////////
+// MCOPY
+
+TEST(InterpreterTest, MCOPY_PreCancun) {
+  RunInterpreterStepTest({
+      .code = {op::MCOPY},
+      .state_after = RunState::kErrorOpcode,
+      .revision = EVMC_SHANGHAI,
+  });
+}
+
+TEST(InterpreterTest, MCOPY) {
+  RunInterpreterStepTest(
+      {
+          .code = {op::MCOPY},
+          .state_after = RunState::kRunning,
+          .pc_before = 0,
+          .pc_after = 1,
+          .gas_before = 1000,
+          .gas_after = 1000 - 6,
```

Comment:
it seems error prone to use the same literals in all tests. maybe we could add a `const` in a scope just for these tests that is used for `pre_existing_gas`, `static_costg` and `dynamic_cost` (for the latter tests) ?

Reply:
In my experience, tests need to be explicit and global constants are vulnerable to be changed in the future, breaking tests or triggering way more work than needed.
The beauty of this code is that you do not need to scroll anywhere to see what it is supposed to be doing. A global constant defined in a 5000+ lines requires unnecessary scrolling.